tutorials/pretrain_openwebtext.md

```diff
@@ -0,0 +1,139 @@
+# Pretrain Llama 2 on OpenWebText
+
+This tutorial will walk you through setting up the OpenWebText dataset and launching the pretraining script.
+
+## What's OpenWebText
+
+[OpenWebText](https://github.com/jcpeterson/openwebtext) is an open-source reproduction of OpenAI's unreleased WebText training dataset, which was originally used to train GPT-2. 
+
+The version that is used here consists of 8M documents and is loaded via the `load_dataset("openwebtext", ...)` function from the [datasets](https://github.com/huggingface/datasets) Python package.
+
+[Please refer to the website hosting the dataset](https://huggingface.co/datasets/Skylion007/openwebtext) for license information.
+
+
+## Prepare OpenWebText for training
+
+
+In order to start pretraining lit-gpt
+on it, you need to read, tokenize, and write the data in binary format.
+
+Do prepare the dataset with the Llama 2 tokenizer, run
+
+```bash
+python scripts/prepare_openwebtext.py \
+  --checkpoint_dir checkpoints/meta-llama/Llama-2-7b-hf/ \
+  --destination_path data/lit-openwebtext
+```
+
+The script will take about 15 min to run.
+
+
+## Pretraining
+
+Running the pretraining script with its default settings requires at least 4 GPUs with 40GB+ each. (However, alternatively, you can train a smaller Pythia-70m on 1 GPU, more information about that further below).
+
+```bash
+python pretrain/openwebtext.py \
+  --devices 4 \
+  --train_data_dir data/lit-openwebtext
+```
+
+The script will save checkpoints periodically to the folder `out/`.
+
+By default, the `pretrain/openwebtext.py` script will pretrain the Llama 2 7B model with FSDP in
+`bfloat16` precision and gradient accumulation.
+
+You can easily change the size of the model by passing a different string to the model name variable
+
+```python
+model_name = "Llama-2-7b-hf"
+```
+
+at the top of this script.
+
+The currently supported model names are contained in the [config.py](https://github.com/Lightning-AI/lit-gpt/lit_gpt/config.py) file. 
+You can either search this file for lines containing "name =" or obtain the list of all supported models programmatically, as follows:
+
+```python
+from lit_gpt.config import configs
+
+for conf in configs:
+    print(conf["name"])
+```
+
+Keep in mind that the original LLaMA training for the 7B model required 83k A100 80GB
+hours (on a bigger dataset). However, for full pretraining on OpenWebText, you'll likely still need access to a cluster.
+
+Once you're in a cluster, you can follow [these instructions](https://lightning.ai/docs/fabric/stable/guide/multi_node/other.html)
```

Comment:
"other" is not representative of the majority here I think. The ones listed there are rather exotic environments.
This might be a better link: https://lightning.ai/docs/fabric/stable/fundamentals/launch.html#launch-on-a-cluster

Reply:
```suggestion
Once you're in a cluster, you can follow [these instructions](https://lightning.ai/docs/fabric/stable/fundamentals/launch.html#launch-on-a-cluster)
```

Reply:
thanks!